packages/@expo/cli/src/start/server/metro/runServer-fork.ts

```diff
@@ -120,6 +135,16 @@ export const runServer = async (
         }),
       });
 
+      // Extend the close method to ensure that we clean up any open connections to
+      // prevent hanging when the server is closed after connecting to clients.
+      const originalClose = httpServer.close.bind(httpServer);
+
+      httpServer.close = (callback?: (err?: Error) => void) => {
+        debug('Closing connections:', connections.length);
+        connections.forEach((conn) => conn.destroy());
+        return originalClose(callback);
```

Comment:
Move the call to originalClose() above destroying the connections so new connections will be rejected. I think server.close() doesn't return anything.